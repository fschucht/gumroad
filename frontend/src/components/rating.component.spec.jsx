import renderer from 'react-test-renderer';
import { Rating } from './rating.component';

describe('Rating', () => {
  function testRating(averageRating) {
    describe(`when the averageRating is ${averageRating}`, () => {
      it(`should render ${Math.ceil(averageRating)} stars`, () => {
        const renderedComponent = renderer.create(
          <Rating
            averageRating={averageRating}
            ratingsCount={100}
          >
            Hello World!
          </Rating>
        ).toJSON()

        expect(renderedComponent).toMatchSnapshot()
      })
    })
  }

  testRating(0)
  testRating(0.25)
  testRating(0.5)
  testRating(0.75)
  testRating(1)
  testRating(1.25)
  testRating(1.5)
  testRating(1.75)
  testRating(2)
  testRating(2.25)
  testRating(2.5)
  testRating(2.75)
  testRating(3)
  testRating(3.25)
  testRating(3.5)
  testRating(3.75)
  testRating(4)
  testRating(4.25)
  testRating(4.5)
  testRating(4.75)
  testRating(5)
})