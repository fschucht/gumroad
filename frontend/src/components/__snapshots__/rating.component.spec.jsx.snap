// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating when the averageRating is 0 should render 0 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    0.0
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 0.5 should render 1 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    0.5
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 0.25 should render 1 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    0.3
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 0.75 should render 1 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    0.8
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 1 should render 1 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    1.0
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 1.5 should render 2 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    1.5
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 1.25 should render 2 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    1.3
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 1.75 should render 2 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    1.8
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 2 should render 2 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    2.0
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 2.5 should render 3 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    2.5
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 2.25 should render 3 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    2.3
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 2.75 should render 3 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    2.8
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 3 should render 3 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    3.0
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 3.5 should render 4 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    3.5
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 3.25 should render 4 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    3.3
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 3.75 should render 4 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    3.8
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 4 should render 4 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    4.0
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 4.5 should render 5 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    4.5
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 4.25 should render 5 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    4.3
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 4.75 should render 5 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    4.8
     (
    100
     Ratings)
  </p>
</div>
`;

exports[`Rating when the averageRating is 5 should render 5 stars 1`] = `
<div
  className="rating"
>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active"
  >
    star.svg
  </svg>
  <svg
    className="rating__star rating__star--active rating__star--last"
  >
    star.svg
  </svg>
  <p
    className="text"
  >
    5.0
     (
    100
     Ratings)
  </p>
</div>
`;
